taws_minimal/tests/cucumber.rs

```diff
@@ -21,280 +19,204 @@ fn main() {
-    world.update_phase(StepType::Given);
-    if maybe_not == "not" {
-        world.constraints[world.phase].add_take_off_constraint(false)
-    } else {
-        world.constraints[world.phase].add_take_off_constraint(true)
-    }
+#[given(expr = "steep approach {maybe} selected")]
+fn given_steep_approach_selected(world: &mut MyWorld, maybe: MaybeParameter) {
+    world.constraints[world.phase].add_steep_approach_constraint(maybe.into());
 }
 
-#[given(regex = r"^go around is ?(not)? selected$")]
-fn go_around(world: &mut MyWorld, maybe_not: String) {
-    world.update_phase(StepType::Given);
-    if maybe_not == "not" {
-        world.constraints[world.phase].add_go_around_constraint(false)
-    } else {
-        world.constraints[world.phase].add_go_around_constraint(true)
-    }
+#[given(expr = "non-precision approach {maybe} selected")]
+fn given_precision_approach_selected(world: &mut MyWorld, maybe: MaybeParameter) {
+    let maybe: bool = maybe.into();
+    world.constraints[world.phase].add_precision_approach_constraint(!maybe);
 }
 
-#[given(regex = r"^non-precision approach is ?(not)? selected$")]
-fn precision_approach(world: &mut MyWorld, maybe_not: String) {
-    world.update_phase(StepType::Given);
+#[given(expr = "take-off {maybe} selected")]
+fn given_take_off(world: &mut MyWorld, maybe: MaybeParameter) {
+    world.constraints[world.phase].add_take_off_constraint(maybe.into());
+}
 
-    // double negation: non-precision is not selected
-    if maybe_not == "not" {
-        world.constraints[world.phase].add_precision_approach_constraint(true);
-    } else {
-        world.constraints[world.phase].add_precision_approach_constraint(false);
-    }
+#[given(expr = "go around {maybe} selected")]
+fn given_go_around(world: &mut MyWorld, maybe: MaybeParameter) {
+    world.constraints[world.phase].add_go_around_constraint(maybe.into());
 }
 
-#[then(regex = r"^(.+) shall ?(not)? be armed$")]
-fn shall_be_armed(world: &mut MyWorld, alert: AlertWrapper, maybe_not: String) {
-    assert_eq!(world.taws.is_armed(alert.into()), !maybe_not.eq("not"));
+#[given(expr = "the height above terrain is {constraint} foot")]
+#[when(expr = "the height above terrain is {constraint} foot")]
+fn given_height_above_terrain(world: &mut MyWorld, height_above_terrain: ConstraintParameter) {
+    let height_above_terrain: Constraint<f64> = height_above_terrain.into();
+
+    let unit = Length::new::<length::foot>(1.0);
+    let height_above_terrain = match height_above_terrain {
+        Constraint::AtLeast(p, a) => Constraint::AtLeast(p, a * unit),
+        Constraint::AtMost(p, a) => Constraint::AtMost(p, a * unit),
+        Constraint::Equal(p, a) => Constraint::Equal(p, a * unit),
+        Constraint::InRange(p, a, b) => Constraint::InRange(p, a * unit, b * unit),
+        Constraint::NotInRange(p, a, b) => Constraint::NotInRange(p, a * unit, b * unit),
+    };
+
+    //let height_above_terrain = height_above_terrain * Length::new::<length::foot>(1.0);
```

Comment:
Why is this commented out but here? If we will need it eventually, please add a short explanation (maximum one sentence) beginning with `// TODO`, otherwise please remove it

Reply:
Was a neat idea to parse the constraints as f64 and unit parameters separate and then define the scalar multiplication for constraints to multiply the constraint with the unit to get a new constraint with unit. But if the unit is negative (e.g. rate of descent) you would have to flip the constraints around (at most -> at least) but uom quantities don't implement the Signed trait (which actually makes sense). So not really practical.

Reply:
I see, good though!